Source/Dafny/Printer.cs

```diff
@@ -1669,11 +1685,21 @@ private bool PrintModeSkipGeneral(Bpl.IToken tok, string fileBeingPrinted)
       }
     }
 
-    void PrintAlternatives(int indent, List<GuardedAlternative> alternatives) {
+    void PrintAlternatives(int indent, List<GuardedAlternative> alternatives, bool startWithLine = true) {
+
       foreach (var alternative in alternatives) {
-        wr.WriteLine();
+        if (startWithLine)
+        {
+          wr.WriteLine();
+        }
+        else
+        {
+          startWithLine = true;
+        }
```

Comment:
Compress these lines by putting the curly braces on the lines with the `if` and the `else`.